### ISSUE TITLE
Powershell dsc ArcGISInstall exception during start calling access denied

### ISSUE THREAD

**Issue by YushaRaiden** [Esri/arcgis-powershell-dsc#126]
Hello,

We have an issue with command Configure-ArcGIS. We use a server on CentOS 7 and we deployed 6 vms with VirtualBox : One on Windows 10 and the others on Widows Server 2016 on this physical server. These VMs are configured with NetBios with suffixe DNS and we installed DSC and ArcGIS module on each machine. We added on TrusdeHosts the ip adress of each machine on the Windows 10 from where we execute the command. And the nodenames on configurationfile.json have also ip adress. But when we try to execute Configure-ArcGIS with the same administrator Credential on each  machine, we receive a message error saying : Except for "Start" call with "1" argument(s) : "Access denied".
Anyone has idea?

Thank you

**Comment by nshampur** [comment]
Are you able to exercise a simple DSC push configuration in this environment

https://docs.microsoft.com/en-us/powershell/dsc/quickstarts/website-quickstart

**Comment by YushaRaiden** [comment]
Hi nshampur,

Thank you for your reply. I'll test this next monday and I shall come back to you quickly to share with you results. 
I' wonder if the host machine from which I launch Configure-ArcGIS is actually the source of my problem.  
More specifically, the login and the password are not the same as other machines.
Could that be a problem to launch Configure-ArcGIS?

Thank you

**Comment by YushaRaiden** [comment]
Hi,

I'm not able to push the config to a remote computer. It's tag as failed

**Comment by shailesh91** [comment]
@YushaRaiden Can you even ping one machine from the other? If so, is PS Remoting enabled considering these are windows 10 VMs?

**Comment by YushaRaiden** [comment]
Hello shailesh91,

Indeed it was my fault because i didn't set files installer in the installer path for all servers Esri.
Sorry. And I have another problème with the installer for Application Request Routing 3.0 . The error generated is [error] Failed to Install Microsoft Application Request Routing `3.0`. All my vm and thier IIS server are signed with openssl by a root certificate and intermediate certificate and a server certificate . I took for example your BaseDeployment-MultiMachine.json (see my configuration joined)
[Mercator-Multimachines.txt](https://github.com/Esri/arcgis-powershell-dsc/files/2906005/Mercator-Multimachines.txt).
To finish my vms was installed in french and after setting with English language by default. Howerver ArcGISWindowsService module is unable to find ArcGIS_WindowsService.strings.psd1 in  C:\Program Files\WindowsPowerShell\Modules\ArcGIS\1.1.4\DscResources\ArcGIS_WindowsService\fr-FR\ .
Sorry, I'm french and not really comfortable in english.
Thank you for help

**Comment by YushaRaiden** [comment]
A few more details: I have checked the product id of the msi packages and they correspond well to what is in the json file. Are there any prerequisites on the LoadBalancer machine? I signed the IIS server with openssl certificates.

**Comment by shailesh91** [comment]
@YushaRaiden 
- We don't support any language other than English as of now. 
- I also noticed that you added a federation block, since it is automatically inferred. For a federated server role you will have to create a different JSON Config. 
- For ARR can you please check if the Product Id matches that of the installer requestRouter_x64.msi.

**Comment by YushaRaiden** [comment]
Thank you for all your answers. Indeed, I was not using the correct version of requestRouter compatible with Windows Server 2016. It works now but I have a problem with the confg-store. On the configuration of one of my two servers I have the following error:  The configuration store location contains server configuration files that are being used by another site. File sharing is done via the virtualbox configuration a tree already created: /Vm/configStore_AGS_Server/arcgisserver/config-store/ .
Do you have any suggestions regarding the configuration of shared directories?

**Comment by shailesh91** [comment]
@YushaRaiden Was the config-store directories already created before the run of the Configure-ArcGIS command?

**Comment by YushaRaiden** [comment]
Please excuse me, sir. The error comes from me since during the second attempt I did not delete the site configuration of the first. After removing the first attempt it worked.
I have problems with the configuration of the servers, the webadaptor and Portal. A priori the services cannot communicate together.
And I feel like the Portal site isn’t creating itself. In fact, all my machines are configured in NetBios with machine name and DNS suffix. For each of the machines, I imported into IIS a root authority certificate, a threaded certificate and a server certificate corresponding to each machine.
All certificates were also imported into the Mmc console as a computer account certificate.
When I test the urls by hand on the server machines by adding port 6443 the site is signed and trusted. However, this is not the case when I test the url by hand of the portal adding the port 7443.
All urls tested in https without ports are properly signed.
I also added in the hosts files of each machine the ip and DNS resolution of all other machines. I did the same thing in winrm’s TrustedHosts. But that doesn’t change my problem.
If you have time to watch, here are the logs with debugswitch. Some parts are in French since the machines were originally installed in this language. I passed them in English.
Thank you for your help. I believe in your solution and I don't I can't make it work.

[Logs_deployment_MultiMachine.txt](https://github.com/Esri/arcgis-powershell-dsc/files/2956254/Logs_deployment_MultiMachine.txt)

**Comment by shailesh91** [comment]
@YushaRaiden can you attach the config file you are using for the deployment?

**Comment by YushaRaiden** [comment]
[Mercator-Multimachines.txt](https://github.com/Esri/arcgis-powershell-dsc/files/2965438/Mercator-Multimachines.txt)

Thank for you reply. Here is the configuration file I use. I noticed that the portal installation doesn’t go all the way. In fact, when I try to finalize it by hand I realize that Firefox does not use the certificates of authority and server certificate that I imported. He seems to be using a signed auto certificate that is not recognized as a trusted authority. 
And when I test the url of the web server (https://ags-mercator3.geomatic.local) it uses the certificate I imported into firefox. However when I add port 7443, Firefox uses a self-signed certificate and does not recognize portal as a trusted site.
I wonder if my problem comes from the error message below.

[AGS-MERCATOR3]:                            [[ArcGIS_Portal]Portalags-mercator3] PortalAdmin UserName:- AdminGeoThales
[AGS-MERCATOR3]:                            [[ArcGIS_Portal]Portalags-mercator3] Configured hostname '' does not match expected value 'ags-mercator3.geomatic.local'. Setting it
[AGS-MERCATOR3]:                            [[ArcGIS_Portal]Portalags-mercator3] hostname.properties file was modified. Need to restart the 'Portal for ArcGIS' service to pick up changes
[AGS-MERCATOR3]:                            [[ArcGIS_Portal]Portalags-mercator3] In Portal DataStore Configured host identifier '' does not match expected value 'ags-mercator3.geomatic.local' or host identifier type '' does not match expected value 'hostname'. Setting it
[AGS-MERCATOR3]:                            [[ArcGIS_Portal]Portalags-mercator3] In Portal DataStore Hostidentifier.properties file was modified. Need to restart the 'Portal for ArcGIS' service to pick up changes.

Finally, I tried to add a Vm with a domain controller so as not to go through the NetBios but I still have the same problem.

Thank you.

**Comment by shailesh91** [comment]
Something is wrong with your network configuration. For example, your FQDN "ags-mercator1.geomatic.local" is not resolving correctly. 
In your logs, I found this in relation to Create site - "The server was unable to access the URL 'http://ags-mercator1.geomatic.local:6080/arcgis/services'. Please check your firewall and security settings. For more information, see the ArcGIS Server help topic Ports used by ArcGIS Server. You can access this topic in the table of contents by navigating to Administering ArcGIS Server > Securing your ArcGIS Server site > Configuring a secure environment for ArcGIS Server."

And the error you are referring to is because of the fact that certificate is in machines local store and not yet imported into the certificate store of the Portal or Server. 

PS. the property you are pointing is not an error. It is just setting the Hostname properties file for portal if it is not set already.

**Comment by YushaRaiden** [comment]
shailesh91,
I checked my network configuration and the ports are well open in Inbound Rules of firewall advanced settingsfor each machines. In the meantime, I replaced the password of the WebAdaptor pfx certificate that had special characters with a password without special characters. I also authorized the ports listed for ArcGIS Server, and Portal for ArcGIS in the Inbound Rules of firewall advanced settings of my ArcGIS Server and Portal for ArcGIS machines.
The arcgis server sites are well created and accessible by the WebAdaptor but Portal always encounters the following error:  
[AGS-MERCATOR3]:                            [[ArcGIS_Portal]Portalags-mercator3] Configured hostname '' does not match expected value 'ags-mercator3.geomatic.local'. Setting it
[AGS-MERCATOR3]:                            [[ArcGIS_Portal]Portalags-mercator3] hostname.properties file was modified. Need to restart the 'Portal for ArcGIS' service to pick up changes
[AGS-MERCATOR3]:                            [[ArcGIS_Portal]Portalags-mercator3] In Portal DataStore Configured host identifier '' does not match expected value 'ags-mercator3.geomatic.local' or host identifier type '' does not match expected value 'hostname'. Setting it

For this new test I modified my configurationfile to set machine name on nodename and all these Is managed with an AD which is on the LoadBalancer macchine. Here my logs and configuration file:  
[Mercator-Multimachines V2.txt](https://github.com/Esri/arcgis-powershell-dsc/files/2977585/Mercator-Multimachines.V2.txt)
[Logs_deployment_MultiMachine0318March.txt](https://github.com/Esri/arcgis-powershell-dsc/files/2977598/Logs_deployment_MultiMachine0318March.txt)

**Comment by shailesh91** [comment]
Looks like your ContentDirectoryLocation might be a problem. Right now it is "\\\\VBOXSVR\\content". It should be something like "\\\\[Fileshare Machine Name 1]\\[Fileshare Name]**\\portalforarcgis\\content\\arcgisportal\\content**". Try appending the bold part. Let me know how it goes.

**Comment by YushaRaiden** [comment]
Please forgive me for wasting your time, I really thought it couldn’t have come from this. I did what you told me (\\\\VBOXSVR\\portalforarcgis\\content\\arcgisportal\\content) and it worked. Really thank you. I just have one last question.  The federated stage doesn’t work.  Do you think this comes from a bad LoadBalancer (ags-mercator7) configuration? The advanced settings of firewall (Inbound rules)? Or other?
Here the message error I have :

VERBOSE: Time taken for configuration job to complete is 0.026 seconds
Perform operation 'Invoke CimMethod' with following parameters, ''methodName' = SendConfigurationApply,'className' = MSFT_DSCLocalConfigurationManager,'namespaceName' = root/Microsoft/Windows/DesiredStateConfiguration'.
Un appel de méthode du Gestionnaire de configuration local est arrivé de l'ordinateur AGS-MERCATOR6 avec le SID utilisateur S-1-5-21-21467652-3102438666-1386676999-500.
[AGS-MERCATOR1] : Gestionnaire de configuration local : [ Début  Définir  ]
[AGS-MERCATOR1]: LCM:  [ Début  Ressource]  [[ArcGIS_Federation]Federate]
[AGS-MERCATOR1]: LCM:  [ Début  Test     ]  [[ArcGIS_Federation]Federate]
[AGS-MERCATOR1]:                            [[ArcGIS_Federation]Federate] Get Portal Token from Deployment 'ags-mercator7.geomatic.local'
[AGS-MERCATOR1]:                            [[ArcGIS_Federation]Federate] Error getting Token for Federation ! Waiting for 1 Minutes to try again

**Comment by shailesh91** [comment]
@YushaRaiden Something is not correct with your load balancer configuration. Your load balancer portal endpoint is not accessible on AGS-MERCATOR1. Try logging into AGS-MERCATOR1 and try accessing portal lb endpoint through a browser window.

**Comment by YushaRaiden** [comment]
Thank you for your advise. I will try this and come to you later

**Comment by YushaRaiden** [comment]
@shailesh91 I tested access to LoadBalancer from ags-mercator1 by broswer Windows and this is what I have : ![BroswerWindows_ConnectionAGSMercator1ToLB](https://user-images.githubusercontent.com/47668200/55066195-89877200-507d-11e9-8365-741bdb989540.png)
I also tested access to LoadBalancer from ags-mercator1 by Firefox broswer and this is what I have : 
![BroswerFirefox_ConnectionAGSMercator1ToLB](https://user-images.githubusercontent.com/47668200/55066568-5c878f00-507e-11e9-9837-a6ccb3333f27.png)
It also appears that on some machines the network discovery deactivates itself. Some forums say to check the services DNS Client, Function Discovery Resource Publication, SSDP Discovery and UPnP Device Host. SSDP Discovery and UPnP Device Host services  were disabled on machine LB and UPnP Device Host service was disabled on other machines.
Are there sharing configurations to be made on the different machines?
